src/components/graph-view.js

```diff
@@ -273,16 +281,23 @@ class GraphView extends React.Component<IGraphViewProps, IGraphViewState> {
       !nextState.componentUpToDate ||
       nextProps.selected !== this.props.selected ||
       nextProps.readOnly !== this.props.readOnly ||
-      nextProps.layoutEngine !== this.props.layoutEngine
+      nextProps.layoutEngine !== this.props.layoutEngine ||
+      nextState.adjusted !== this.state.adjusted ||
+      nextState.mounted !== this.state.mounted ||
+      nextState.firstAdjustment !== this.state.firstAdjustment
```

Comment:
why do we need all these extra checks? can we just add a callback on layout engine update and reset the graph state whenever the layout is changed ?

Reply:
Good catch, thanks, I added a check for the new state variable to ensure I see the effects of their updates, but the most salient part is captured by the check on `layoutEngine`.

Reply:
Actually, `firstAdjustment` is necessary since it is used in the `render` method.